app/lib/send_grid_client.rb

```diff
@@ -17,6 +17,52 @@ def add_user(user)
     Rails.logger.error e.backtrace.join("\n")
   end
 
+  # Sends a transactional email with a template, based on the passed in options hash.
+  #
+  # @param options [Hash] Sample options hash:
+  #   {
+  #     to: 'tom@gmail.com',
+  #     from: 'frank@oc.com',
+  #     subject: 'Hello',
+  #     user_count: 5,
+  #     body: 'Hello world!',
+  #     template_id: 'asdfas-123123-asdfasf-2qwerqwe'
+  #   }
+  # @return [SendGrid::Response] Includes a body and status_code
+  # @see https://github.com/sendgrid/sendgrid-ruby/blob/7c8973f7b8de51a98dce73a52138791b19434573/USE_CASES.md#with-mail-helper-class
+  #
+  def send_email_with_template(options={})
+    Rails.logger.info("Sending email to #{options[:to]} at #{Time.current} re: #{options[:subject]}")
+
+    mail = {
+      "personalizations" => [
+        {
+          "to" => [
+            {
+              "email" => options[:to]
+            }
+          ],
+          "subject" => options[:subject],
+          "template_id" => options[:transactional_template_id]
+        }
+      ],
+      "from" => {
+        "email" => options[:from]
+      },
+      "content" => [
+        {
+          "type" => "text/plain",
+          "value" => options[:body]
+        }
+      ]
+    }
+
+    @sendgrid.client.mail._("send").post(request_body: mail)
+  rescue => e
+    Rails.logger.error "Failed to send email to #{options[:to]} re: #{options[:subject]} due to: #{e}"
+    Rails.logger.error e.backtrace.join("\n")
```

Comment:
Resolved the `user_count` issue: https://github.com/OperationCode/operationcode_backend/pull/92/commits/6d5668d581e73e602f189b55dc54ad2fa3771f84